packages/amplify-cli-core/src/category-interfaces/categoryBaseGenSchema.ts

```diff
@@ -0,0 +1,123 @@
+
+  private getSchemaFileNameForType(typeName: string): string {
+    return `${typeName}.schema.json`;
+  }
+
+  private getTypesSrcRootForSvc(svcName: string): string {
+    return `${this.TYPES_SRC_ROOT}/${svcName}-user-input-types.ts`;
+  }
+
+  private getSvcFileAbsolutePath(svcName: string): string {
+    return path.resolve(this.getTypesSrcRootForSvc(svcName));
+  }
+
+  private printWarningSchemaFileExists() {
+    printer.info('The interface version must be bumped after any changes.');
+    printer.info(`Use the ${this.OVERWRITE_SCHEMA_FLAG} flag to overwrite existing versions`);
+    printer.info('Skipping this schema');
+  }
+
+  private printSuccessSchemaFileWritten(typeName: string) {
+    printer.info(`Schema written for type ${typeName}.`);
+  }
+
+  constructor(typeDefs: TypeDef[]) {
+    this.serviceTypeDefs = typeDefs;
+  }
+
+  public generateJSONSchemas(): string[] {
+    const force = process.argv.includes(this.OVERWRITE_SCHEMA_FLAG);
+    const generatedFilePaths: string[] = [];
+
+    // schema generation settings. see https://www.npmjs.com/package/typescript-json-schema#command-line
+    const settings: PartialArgs = {
+      required: true,
+    };
+
+    for (const typeDef of this.serviceTypeDefs) {
+      //get absolute file path to the user-input types for the given service
+      const svcAbsoluteFilePath = this.getSvcFileAbsolutePath(typeDef.service);
+      printer.info(svcAbsoluteFilePath);
+      //generate json-schema from the input-types
+      const typeSchema = buildGenerator(getProgramFromFiles([svcAbsoluteFilePath]), settings)?.getSchemaForSymbol(typeDef.typeName);
+      //save json-schema file for the input-types. (used to validate cli-inputs.json)
+      const outputSchemaFilePath = path.resolve(
+        path.join(this.SCHEMA_FILES_ROOT, typeDef.service, this.getSchemaFileNameForType(typeDef.typeName)),
+      );
+      if (!force && fs.existsSync(outputSchemaFilePath)) {
+        this.printWarningSchemaFileExists();
+        return generatedFilePaths;
+      }
+      fs.ensureFileSync(outputSchemaFilePath);
+      fs.writeFileSync(outputSchemaFilePath, JSON.stringify(typeSchema, undefined, 4));
+      //print success status to the terminal
+      this.printSuccessSchemaFileWritten(typeDef.typeName);
+      generatedFilePaths.push(outputSchemaFilePath);
+    }
+    return generatedFilePaths;
+  }
+}
+
+//Read Schema, Validate and return Typescript object.
+export class CLIInputSchemaValidator {
+  _category: string;
+  _service: string;
+  _schemaFileName: string;
+  _ajv: Ajv.Ajv;
+
+  constructor(service: string, category: string, schemaFileName: string) {
+    this._category = category;
+    this._service = service;
+    this._schemaFileName = schemaFileName;
+    this._ajv = new Ajv();
+  }
+
+  async getUserInputSchema() {
+    try {
+      return await import(
+        `@aws-amplify/amplify-category-${this._category}/resources/schemas/${this._service}/${this._schemaFileName}.schema.json`
```

Comment:
Can this be defined in File-Path utils file.  (category, service, fileName)=> schemaFileObject

Reply:
done